### PR TITLE
feat(observe): implement observe connector

### PR DESCRIPTION
**Pull request by GALLLASMILAN**
### Which issue(s) does this pull-request address?

Closes: https://github.ibm.com/Incubation/bee-observe/issues/8

### Description

I implemented the `bee-observe-connector` directly into the framework. We have to to wait for the [bee-observe](https://github.ibm.com/Incubation/bee-observe) and [bee-observe-connector](https://github.ibm.com/Incubation/bee-observe-connector) to be public

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included **No code changes**
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -144,3 +144,6 @@ dist
 # Code interpreter data
 /examples/tmp/code_interpreter/*
 /examples/tmp/local/*
+
+# Observe data
+trace.json
```
Comment (Tomas2D) [inline .gitignore:149]: Use `/examples/tmp/observability/trace.json` to save the trace.

And add this line to the `.gitignore`

```
/examples/tmp/observability/*
```

File: README.md [hunk 2]
```diff
@@ -71,6 +71,8 @@ To run this example, be sure that you have installed [ollama](https://ollama.com
 
 ➡️ See a more [advanced example](./examples/agents/bee.ts).
 
+➡️ See the [Observe API / mlflow example](./examples/agents/bee_observe.ts).
```
Comment (Tomas2D) [inline README.md:74]: `MLFlow`

File: README.md [hunk 3]
```diff
@@ -99,6 +101,11 @@ To run this example, be sure that you have installed [ollama](https://ollama.com
 4. Start the code interpreter `yarn run infra:start-code-interpreter`.
 5. Start the agent `yarn run start:bee` (it runs ./examples/agents/bee.ts file).
 
+#### Observe / mlflow
```
Comment (Tomas2D) [inline README.md:104]: `MLFlow` (add link to the MLFlow)

File: README.md [hunk 4]
```diff
@@ -99,6 +101,11 @@ To run this example, be sure that you have installed [ollama](https://ollama.com
...
+
+1. Start all services related to Observe `yarn infra:start-observe`
+2. Start the agent using the observe and mlflow `yarn start:observe` (it runs ./examples/agents/bee_observe.ts file).
```
Comment (Tomas2D) [inline README.md:107]: add link to the file

File: docker-compose.yml [hunk 5]
```diff
@@ -23,6 +23,64 @@ services:
       interval: 10s
       retries: 10
       start_period: 30s
+  mongo:
+    image: mongo:7
```
Comment (Tomas2D) [inline docker-compose.yml:27]: I would pin a version.

File: docs/templates.md
Comment (Tomas2D) [inline docs/templates.md:1]: I will fix this on the main branch.

File: package.json [hunk 6]
```diff
@@ -154,6 +156,8 @@
     "@types/node": "^20.16.1",
     "@types/object-hash": "^3.0.6",
     "@types/turndown": "^5.0.5",
+    "bee-agent-framework": "^0.0.26",
```
Comment (Tomas2D) [inline package.json:159]: Remove.

File: package.json [hunk 7]
```diff
@@ -154,6 +156,8 @@
...
+    "bee-observe-connector": "^0.0.2",
```
Comment (Tomas2D) [inline package.json:160]: This is a peer dependency and a dev dependency.

File: examples/agents/bee_observe.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            maxIterations: 20,
+          },
+        },
+      )
+      .middleware(
+        createObserveConnector({
+          api: beeObserveApiSetting,
+          cb: async (err, data) => {
+            if (err) {
+              console.log(ObserveError.ensure(err).explain());
```
Comment (Tomas2D) [inline examples/agents/bee_observe.ts:46]: use reader

File: infra/observe/entrypoint.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# Create experiment directory and ensure it has the correct permissions
```
Comment (Tomas2D) [inline infra/observe/entrypoint.sh:2]: add `set -e` to fail fast

File: package.json [hunk 10]
```diff
@@ -61,8 +61,10 @@
     "ts:check": "tsc --noEmit && tsc -p tsconfig.examples.json --noEmit",
     "start": "tsx --tsconfig tsconfig.examples.json",
     "start:bee": "yarn start -- examples/agents/bee.ts",
+    "start:observe": "yarn start -- examples/agents/bee_observe.ts",
```
Comment (Tomas2D) [inline package.json:64]: `start:bee_observe`

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: infra/observe/entrypoint.sh (hunk 9 above)
Comment (GALLLASMILAN) [reply infra/observe/entrypoint.sh:5]: done

**Comment by Tomas2D** [comment]
Let's move this to https://github.com/i-am-bee/bee-agent-framework-starter.

**Pull request closed without merging**